proto/provenance/metadata/v0/scope.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+
+package scope;
+
+option go_package          = "github.com/provenance-io/provenance/x/metadata/legacy/v039";
+option java_package        = "io.provenance.metadata.v0";
+option java_multiple_files = true;
+
+import "provenance/metadata/v0/contract.proto";
+import "provenance/metadata/v0/types.proto";
+
+import "google/protobuf/descriptor.proto";
+
+//
+// Overview
+//
+//   When memorializing a contract the only pieces that matter are the results and facts.  We scope these inside the
+//   contract_group structure to represent a scope around this information and keep the controlling parties (recitals)
+//   attached which prevents a co-mingling of the rights to change/update these records.  The proof submitted to record
+//   the facts is part of the readset (the submitted Contract package) and is not important now that the information has
+//   been recorded.  If the source is required it can be pulled from the ReadSet and referenced (or any of the members
+//   that stored it under the associated URI.
+
+// Event references in relation to scope
+message Event {
+  types.UUID group_uuid     = 1;
+  types.UUID execution_uuid = 2;
+}
+
+// A ContractScope is a context that binds multiple p8e working together to create a set of facts managed a defined
+// set of affiliates.
+message Scope {
+  // Unique ID for building up ProvenanceReferences
+  types.UUID uuid = 1;
+  // Required parties on this scope.  These are union with any contract specific recital requirements
+  repeated contract.Recital parties = 2;
+  // Each contract memorialized in this scope is stored in this list of contract_group.  The list of groups is keyed
+  // as a hashmap based on the contract_spec_ref as a unique key.
+  repeated scope.RecordGroup record_group = 3;
+  // Last event that resulted in scope changes.
+  scope.Event last_event = 4;
```

Comment:
will need value_owner here at some point - with whatever type is standard for representing addresses in proto.

Reply:
should have that in the `v1` protos ...